File: py/test_money.py
import unittest
from money import Money
from portfolio import Portfolio

class TestMoney(unittest.TestCase):

    def test_multiplicationInEuros(self):
        tenEuros = Money(10, "EUR")
        twentyEuros = Money(20, "EUR") 
        self.assertEqual(twentyEuros, tenEuros.times(2))

    def test_division(self):
        originalMoney = Money(4002, "KRW")
        actualMoneyAfterDivision = originalMoney.divide(4)
        expectedMoneyAfterDivision = Money(1000.5, "KRW")
        self.assertEqual(expectedMoneyAfterDivision, actualMoneyAfterDivision)

    def test_addtion(self):
        fiveDollars = Money(5, "USD")
        tenDollars = Money(10, "USD")

        fifteenDollars = Money(15, "USD")
        portfolio = Portfolio()
        portfolio.add(fiveDollars, tenDollars)
        self.assertEqual(fifteenDollars, portfolio.evaluate("USD"))

    def test_additionOfDollarsAndEuros(self):
        fiveDollars = Money(5, "USD")
        tenEuros = Money(10, "EUR")
        portfolio = Portfolio()
        portfolio.add(fiveDollars, tenEuros)
        expectedValue = Money(17, "USD")
        actuallValue = portfolio.evaluate("USD")
        self.assertEqual(
                expectedValue, actuallValue, "%s != %s" % (expectedValue, actuallValue)
            )

if __name__ == '__main__':
    unittest.main()

File: js/test_money.js
const assert = require('assert');

class Money {
	constructor(amount, currency) {
		this.amount = amount;
		this.currency = currency;
	}

	times(multiplier) {
		return new Money(this.amount*multiplier, this.currency);
	}

	divide(divisor) {
		return new Money(this.amount/divisor, this.currency);
	}
}

//test-1-multiplication
let fiveDollars = new Money(5, "USD");
let tenDollars = new Money(10, "USD");
assert.deepStrictEqual(fiveDollars.times(2), tenDollars);

//test-2-multiplication
let tenEuros = new Money(10, "EUR");
let twentyEuros = new Money(20, "EUR");
assert.deepStrictEqual(tenEuros.times(2), twentyEuros);

//test-3-division
let originalMoney = new Money(4002, "KRW")
let actualMoneyAfterDivision = originalMoney.divide(4)
let expectedMoneyAfterDivision = new Money(1000.5, "KRW")
assert.deepStrictEqual(actualMoneyAfterDivision, expectedMoneyAfterDivision)
